### PR TITLE
Couple of fixes to LLVM AOT compiler on Windows x64.

### PR DESCRIPTION
**Pull request by lateralusX**
- Enable MSVC objects on LLVM >= 6.00

- Always used PIC relocation-model on Windows x64 due to LLVM problem with
static triggering incorrect ADDR32 relocation entries in 64-bit objects.

- Add CodeView symbol format instead of DWARF for Windows x64.

- Fix LLVM outfile when doing static link to not end up in temp directory.

- Only add /DEFAULTLIB when using MSVC and not doing static compile. For static compile the needed libraries must be added when objects get's linked and since the needed libraries are c-runtime libraries, should be transparent for most applications consuming the object files.

### CODE REVIEW

**Comment by lateralusX** [comment]
@monojenkins rebuild failed

File: mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -9522,8 +9527,8 @@ emit_llvm_file (MonoAotCompile *acfg)
 	g_string_append_printf (acfg->llc_args, " -no-x86-call-frame-opt");
 #endif
 
-#if ( defined(TARGET_MACH) && defined(TARGET_ARM) ) || defined(TARGET_ORBIS)
-	/* ios requires PIC code now */
+#if ( defined(TARGET_MACH) && defined(TARGET_ARM) ) || defined(TARGET_ORBIS) || defined(TARGET_X86_64_WIN32_MSVC)
+
```
Comment (vargaz) [inline mono/mini/aot-compiler.c]: Extra empty line.

File: mono/mini/aot-compiler.c [hunk 2]
```diff
@@ -13471,7 +13482,11 @@ emit_aot_image (MonoAotCompile *acfg)
 			acfg->tmpbasename = g_build_filename (temp_path, "temp", NULL);
 			acfg->tmpfname = g_strdup_printf ("%s.s", acfg->tmpbasename);
 			acfg->llvm_sfile = g_strdup_printf ("%s-llvm.s", acfg->tmpbasename);
-			acfg->llvm_ofile = g_strdup_printf ("%s-llvm." AS_OBJECT_FILE_SUFFIX, acfg->tmpbasename);
+
+			if (acfg->aot_opts.static_link)
+				acfg->llvm_ofile = g_strdup_printf (acfg->aot_opts.llvm_outfile);
```
Comment (vargaz) [inline mono/mini/aot-compiler.c]: Why is this needed ?

Comment (lateralusX) [reply mono/mini/aot-compiler.c]: If I specify the out file for the LLVM object (--llvm-outfile), it will still be generated in the temp folder using the tempbase name. If I specify the out file for the none LLVM object (--outfile) it will be generated as specified by the argument. For none static build this is not a problem since the shared library will be the outfile (the rest will only be temporary artifacts anyway), but for static build, it should be placed as specified in llvm-outfile.

File: mono/mini/aot-compiler.c (hunk 1 above)
Comment (lateralusX) [reply mono/mini/aot-compiler.c]: Fixed.

**Comment by lateralusX** [comment]
@monojenkins rebuild failed

**Comment by lateralusX** [comment]
@vargaz good to go?

File: mono/mini/aot-compiler.c (hunk 2 above)
Comment (vargaz) [reply mono/mini/aot-compiler.c]: We are using these options in ios in static mode, wondering why that works then.

Comment (lateralusX) [reply mono/mini/aot-compiler.c]: OK, that is strange since it will then end up in a temp folder using the default naming schema. Lets do like this, I can keep that change downstream for the platform that currently needs it and remove it from this PR for now to reduce risk of regression.

Comment (lateralusX) [reply mono/mini/aot-compiler.c]: Removed that change from this PR.

**Comment by lateralusX** [comment]
@monojenkins rebuild failed

**Comment by lateralusX** [comment]
@vargaz anything more to adjust on this PR?
